cli/nodeutil.ts

```diff
@@ -174,6 +174,224 @@ export function needsGitCleanAsync() {
         })
 }
 
+export async function getDefaultBranchAsync(): Promise<string> {
+    const b = await gitInfoAsync(["symbolic-ref", "--short", "refs/remotes/origin/HEAD"], undefined, true);
+    if (!b)
+        throw Util.userError("no git remote branch found");
+    return b.replace(/^origin\//, "");
+}
+
+export async function getCurrentBranchNameAsync(): Promise<string> {
+    const b = await gitInfoAsync(["rev-parse", "--abbrev-ref", "HEAD"], undefined, true);
+    if (!b)
+        throw Util.userError("no git local branch found");
+    return b;
+}
+
+export async function getGitHubTokenAsync(): Promise<string> {
```

Comment:
is this tested on wsl and windows?

Reply:
Tested in bash and powershell. Each of us will need to test in our respective dev environments. I'm unable to test the complete matrix.